test/blocksDestructivePushes.js

```diff
@@ -0,0 +1,146 @@
+
+		const mockRepoNoBPR = {
+			defaultBranchRef: {
+				name: 'main',
+				branchProtectionRule: null,
+			},
+		};
+
+		const result2 = metric.extract(mockRepoNoBPR);
+		t.equal(result2, null, 'should return null when branchProtectionRule is null');
+
+		t.end();
+	});
+
+	t.test('should handle missing defaultBranchRef', (t) => {
+		const mockRepoMissingRef = {};
+
+		const result = metric.extract(mockRepoMissingRef);
+		t.equal(result, null, 'should return null when defaultBranchRef is missing entirely');
+		t.end();
+	});
+};
+
+const testBlockingCases = (t) => {
+	const metric = Metrics.BlocksDestructivePushes;
+
+	t.test('should return true when destructive pushes are blocked', (t) => {
+		const mockRepo = {
+			defaultBranchRef: {
+				name: 'main',
+				branchProtectionRule: {
+					allowsForcePushes: false,
+					allowsDeletions: false,
+				},
+			},
+		};
+
+		const result = metric.extract(mockRepo);
+		t.equal(result, true, 'should return true when both force pushes and deletions are disabled');
+		t.end();
+	});
+
+	t.test('should handle undefined as falsy', (t) => {
+		const mockRepo = {
+			defaultBranchRef: {
+				name: 'main',
+				branchProtectionRule: {
+					// allowsForcePushes and allowsDeletions are undefined
+					requiredStatusChecks: [],
+				},
+			},
+		};
+
+		const result = metric.extract(mockRepo);
+		t.equal(result, true, 'should return true when allowsForcePushes and allowsDeletions are undefined (treated as false)');
+		t.end();
+	});
+};
+
+const testAllowingCases = (t) => {
+	const metric = Metrics.BlocksDestructivePushes;
+
+	t.test('should return false when force pushes are allowed', (t) => {
+		const mockRepo = {
+			defaultBranchRef: {
+				name: 'main',
+				branchProtectionRule: {
+					allowsForcePushes: true,
+					allowsDeletions: false,
+				},
+			},
+		};
+
+		const result = metric.extract(mockRepo);
+		t.equal(result, false, 'should return false when force pushes are allowed');
+		t.end();
+	});
+
+	t.test('should return false when deletions are allowed', (t) => {
+		const mockRepo = {
+			defaultBranchRef: {
+				name: 'main',
+				branchProtectionRule: {
+					allowsForcePushes: false,
+					allowsDeletions: true,
+				},
+			},
+		};
+
+		const result = metric.extract(mockRepo);
+		t.equal(result, false, 'should return false when deletions are allowed');
+		t.end();
+	});
+
+	t.test('should return false when both are allowed', (t) => {
+		const mockRepo = {
+			defaultBranchRef: {
+				name: 'main',
+				branchProtectionRule: {
+					allowsForcePushes: true,
+					allowsDeletions: true,
+				},
+			},
+		};
+
+		const result = metric.extract(mockRepo);
+		t.equal(result, false, 'should return false when both force pushes and deletions are allowed');
+		t.end();
+	});
+};
+
+// Execute the tests
+test('BlocksDestructivePushes metric', (t) => {
+	testBasicConfiguration(t);
+	testNullCases(t);
+	testBlockingCases(t);
+	testAllowingCases(t);
+	t.end();
```

Comment:
fwiw i think it'd also be fine to inline all of these functions as a `t.test`, and nest all the subtests under it :-)